src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -143,6 +143,13 @@ public virtual bool RunInMemory
         [ConfigurationEntry("Indexing.Analyzers.NGram.MaxGram", ConfigurationEntryScope.ServerWideOrPerDatabase)]
         public int MaxGram { get; set; }
 
+        [Description("Managed allocations limit, which an index will stop and complete the current batch")]
+        [DefaultValue(512)]
```

Comment:
Why do you think `512` is the golden mean? I think that will drastically reduce the batch sizes for those who have large instances. 

I'm for limiting it but I have some doubts regarding the default value. Why not to limit the batch size (number of docs) or indexing batch size by default instead?

Reply:
The issue is that we saw a customer that had a LOT of allocations in the index.
That meant that we started seeing GC pauses as a result and the cluster become unstable.

Reply:
I understand the reasoning of limiting that and I'm for it. But I have some doubts that 512 is too low. Especially on stronger machines.

As I know @karmeli87 talked to @ppekrol yesterday and there was an idea of setting this to `null` by default and introducing a new performance hint where we could explain the pros and cons of large indexing batches and point to 3 options that we have so a user can limit an indexing batch size.

Reply:
I not in favor for additional toggles here. 
I will run the indexing benchmark + SO import, so we will have some numbers around this change.

Reply:
At least we should put it as a separate position in indexing performance batch stats (and expose it in the Studio) so it will be easier to trace down.

Reply:
I agree that we want to see that in the stats.
Take into account that we also have to deal with multiple indexes at the same time here.
So the 512MB limit is per index, not global.

That said, the issue is that we want to avoid accumulating garbage during indexing. Because we want to reduce the amount of GC pauses.

Reply:
So I ran the indexing benchmark and it is too small to show any difference. So I used the SO dump + indexes.
Imported all users and then deployed the indexes (only 2 out of 6 indexes are relevant)

with `null` - 3:01 min
with `512MB` - 5:30 min

ran it few time (on my machine) and got more or less same results as above.
I guess those results are expected.